shiny/session/_session.py

```diff
@@ -242,72 +243,86 @@ def verify_state(expected_state: ConnectionState) -> None:
             if conn_state != expected_state:
                 raise ProtocolError("Invalid method for the current session state")
 
-        try:
-            await self._send_message(
-                {"config": {"workerId": "", "sessionId": str(self.id), "user": None}}
-            )
+        with contextlib.ExitStack() as stack:
+            try:
+                await self._send_message(
+                    {
+                        "config": {
+                            "workerId": "",
+                            "sessionId": str(self.id),
+                            "user": None,
+                        }
+                    }
+                )
 
-            while True:
-                message: str = await self._conn.receive()
-                if self._debug:
-                    print("RECV: " + message, flush=True)
+                while True:
+                    message: str = await self._conn.receive()
+                    if self._debug:
+                        print("RECV: " + message, flush=True)
 
-                try:
-                    message_obj = json.loads(
-                        message, object_hook=_utils.lists_to_tuples
-                    )
-                except json.JSONDecodeError:
-                    warnings.warn("ERROR: Invalid JSON message", SessionWarning)
-                    return
+                    try:
+                        message_obj = json.loads(
+                            message, object_hook=_utils.lists_to_tuples
+                        )
+                    except json.JSONDecodeError:
+                        warnings.warn("ERROR: Invalid JSON message", SessionWarning)
+                        return
 
-                if "method" not in message_obj:
-                    self._send_error_response("Message does not contain 'method'.")
-                    return
+                    if "method" not in message_obj:
+                        self._send_error_response("Message does not contain 'method'.")
+                        return
 
-                async with lock():
+                    async with lock():
 
-                    if message_obj["method"] == "init":
-                        verify_state(ConnectionState.Start)
+                        if message_obj["method"] == "init":
+                            verify_state(ConnectionState.Start)
 
-                        conn_state = ConnectionState.Running
-                        message_obj = typing.cast(ClientMessageInit, message_obj)
-                        self._manage_inputs(message_obj["data"])
+                            # When a reactive flush occurs, flush the session's outputs,
+                            # errors, etc. to the client.
+                            unreg = on_flushed(self._flush)
```

Comment:
Can you add a comment making it explicit that this is `reactive._core.on_flushed`? I know the comment mentions "reactive flush", but I still was confused for a while and thought this was the `self.on_flushed` method.